Tasks/PerformActionOnMachineGroup/AzureHelper.ps1

```diff
@@ -60,6 +60,32 @@ function Restart-MachineInProvider
 
 function Initialize-AzureHelper
 {
+    Write-Verbose "Entering in azure-initializer" -Verbose
+
     Switch-AzureMode AzureResourceManager
-    Write-Verbose "Switched to AzureResourceManager" -Verbose
+    if($machineGroup.ProviderDataList.Count -gt 0)
+    {
+        $providerDataName = $machineGroup.ProviderDataList[0].Name
+        Write-Verbose "ProviderDataName : $providerDataName" -Verbose
+        $providerData = Get-ProviderData -ProviderDataName $providerDataName -Connection $connection
+        $subscriptionName = $providerData.Properties.GetProperty("SubscriptionName")     
+        $username = $providerData.Properties.GetProperty("Username")
+        $password = $providerData.Properties.GetProperty("Password")
+        Write-Verbose "SubscriptionName : $subscriptionName" -Verbose
+        Write-Verbose "Username : $username" -Verbose
+        $securePassword = ConvertTo-SecureString $password -AsPlainText -Force
+        $psCredential = New-Object System.Management.Automation.PSCredential ($username, $securePassword)
+        $azureAccount = Add-AzureAccount -Credential $psCredential
```

Comment:
Do you want to capture the error in an error variable and display the error here, while doing add-azureaccount